docs/references/fluxyaml-config-files.md

```diff
@@ -1,229 +1,268 @@
+but records changes by running commands as given in the `.flux.yaml`.
 
-### Combining generators, updaters and raw manifest files
-
-The `.flux.yaml` files support including multiple generators and updaters. Here is an example combining multiple
-generators:
+This is how a command-updated `.flux.yaml` looks:
 
 ```yaml
 version: 1
 commandUpdated:
   generators:
-    - command: kustomize build .
-    - command: helm template ../charts/mychart -f overrides.yaml
+  - command: generator_command
+  updaters:
+  - containerImage:
+      command: image_updater_program
+    policy:
+      command: policy_updater_program
 ```
 
-The generators/updaters will simply be executed in the presented order (top down). Flux will merge their output.
+The `updaters` section is particular to command-updated
+configuration. It contains an array of updaters, each of which gives a
+command for updating container images, and a command for updating
+policies (policy controls how automated updates should be applied to a
+resource; these appear as annotations in generated manifests).
 
-Flux supports both generated manifests and raw manifests tracked in the same repository. If Flux doesn't find a
-configuration file associated to a target directory, Flux will inspect it in search for raw YAML manifest files.
+When asked to update a resource, fluxd will run execute the
+appropriate variety of command for *each* entry in `updaters:`. For
+example, when updating an image, it will execute the command under
+`containerImage`, for each updater entry, in turn.
 
-### The `patchUpdated` configuration variant
+Usually updates come in batches -- e.g., updating the same container
+image in several resources -- so the commands will likely be run
+several times.
 
-We mentioned before that, while it is simple for users to provide generator commands, matching updater commands are
-harder to construct. To improve the situation, Flux provides a different configuration variant: `patchUpdated`.
-`patchUpdated` configurations store the modifications from Flux into a
-[YAML merge patch file](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/strategic-merge-patch.md)
-and implicitly apply them to the resources printed by the `generators`.
+### Execution context of commands
 
-Here is an example, allowing to deploy a [Helm chart without a Tiller installation](https://medium.com/@jeroen.rosenberg/using-helm-charts-without-tiller-f1588bc1f0c4)
+`generators` and `updaters` are run in a POSIX shell inside the fluxd
+container. This means that the executables mentioned in commands must
+be available in the [Flux container
+image](https://github.com/fluxcd/flux/blob/master/docker/Dockerfile.flux).
 
-```yaml
-version: 1
-patchUpdated:
-  generators:
-    - command: helm template ../charts/mychart -f overrides.yaml
-  patchFile: flux-patch.yaml
-```
+Flux currently includes `kustomize` and basic Unix shell tools. If the
```

Comment:
```suggestion
Flux currently includes `kustomize`, `helm` and basic Unix shell tools. If the
```

Reply:
Ignore this, we don't include the binary in the image (we only use it for testing).